redaxo/src/addons/users/lib/role.php

```diff
@@ -35,14 +35,28 @@ class rex_user_role implements rex_user_role_interface
      *
      * @param array $params Params
      */
-    private function __construct(array $params)
+    private function __construct($roles)
     {
-        foreach ([rex_perm::GENERAL, rex_perm::OPTIONS, rex_perm::EXTRAS] as $key) {
-            $perms = $params[$key] ? explode('|', trim($params[$key], '|')) : [];
-            $this->perms = array_merge($this->perms, $perms);
-            unset($params[$key]);
+        foreach ($roles as $role_key => $role) {
+            $params = $role;
+            foreach ([rex_perm::GENERAL, rex_perm::OPTIONS, rex_perm::EXTRAS] as $key) {
+                $perms = $params[$key] ? explode('|', trim($params[$key], '|')) : [];
```

Comment:
hatten wir für das handling von multi-values in einer db-col extra funktionen damit man solche dinge nicht mehr manuell machen muss?